app/models/certificate.server.ts

```diff
@@ -0,0 +1,64 @@
+import type { Certificate } from '@prisma/client';
+
+import { prisma } from '~/db.server';
+
+export type { Certificate } from '@prisma/client';
+
+export async function getCertificateByUsername(username: Certificate['username']) {
+  return prisma.certificate.findUnique({ where: { username } });
+}
+
+export async function createCertificate(
+  username: Certificate['username'],
+  subject: Certificate['subject'],
+  certificate: Certificate['certificate'],
+  orderUrl: Certificate['orderUrl'],
+  privateKey: Certificate['privateKey']
+) {
+  // Set expiration date 90 days from now
+  let validTo = new Date();
+  validTo.setDate(validTo.getDate() + 90);
+
+  return prisma.certificate.create({
+    data: {
+      username,
+      subject,
+      certificate,
+      orderUrl,
+      privateKey,
+      validTo,
+    },
+  });
+}
+
+export async function updateCertificateByUsername(
+  username: Certificate['username'],
+  subject?: Certificate['subject'],
+  certificate?: Certificate['certificate'],
+  orderUrl?: Certificate['orderUrl'],
+  privateKey?: Certificate['privateKey'],
+  validFrom?: Certificate['validFrom']
+) {
+  // If validFrom is changed, set validTo to 90 days from validFrom
+  let validTo;
+  if (validFrom) {
+    validTo = validFrom;
+    validTo.setDate(validTo.getDate() + 90);
+  }
+  return prisma.certificate.update({
+    where: { username },
+    data: {
+      username,
+      subject,
+      certificate,
+      orderUrl,
+      privateKey,
+      validFrom,
+      validTo,
+    },
+  });
+}
```

Comment:
From what I recall with `update`, you always need to pass the original values right?

Reply:
If the value is `undefined`, Prisma will know to leave that alone. Not `null`, by the way.

Reply:
should we refrain from updating the `unchanged` values?

Reply:
Good question!

Reply:
@Ririio Well, it will just effectively do nothing, so it shouldn't cause problems either way.

Reply:
of course lmfao, I was just wondering